### PR TITLE
Create relocated package for spark-connect-rpc

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

This PR introduces a new module `kyuubi-relocated-spark-connect-rpc`, which forks [Spark Connect proto files](https://github.com/apache/spark/tree/5b2d2149b615acdd8730547a1f24c2b637222545/sql/connect/common/src/main/protobuf) and shifts the package name from `org.apache.spark.connect.proto` to `org.apache.kyuubi.shaded.spark.connect.proto`, this is a precondition for Kyuubi to support Spark Connect

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-relocated-spark-connect-rpc/src/main/protobuf/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+proto files are copied from
+https://github.com/apache/spark/tree/5b2d2149b615acdd8730547a1f24c2b637222545/sql/connect/common/src/main/protobuf
+
+and with one additional change in each proto file
```
Comment (cxzl25) [inline kyuubi-relocated-spark-connect-rpc/src/main/protobuf/README.md:4]: We can provide an automatic script or command for batch modification in the future.

Comment (pan3793) [reply kyuubi-relocated-spark-connect-rpc/src/main/protobuf/README.md:4]: The cost of manual modification is not high, but we do need CI to ensure the correctness of the package name

**Review by cxzl25 — APPROVED** [review]

**Comment by pan3793** [comment]
Merged to master

File: kyuubi-relocated-spark-connect-rpc/src/main/protobuf/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+proto files are copied from
+https://github.com/apache/spark/tree/5b2d2149b615acdd8730547a1f24c2b637222545/sql/connect/common/src/main/protobuf
```
[the remark this answers is not in the document]
Comment (pan3793) [reply kyuubi-relocated-spark-connect-rpc/src/main/protobuf/README.md:2]: > It appears that the proto file has been forked from the spark 4.0.0-SNAPSHOT

Yes, I think there will be no major changes before 4.0.0 GA, and I will make a sync after Spark 4.0.0-preview2 is available.

> Which version do we intend to initially support?

Spark 4.0.0. I suppose the first version will be merged into Kyuubi master after `branch-1.10` cut out, and Spark 4.0.0-preview2 or 4.0.0 will be available at that time.

**Comment by ulysses-you** [comment]
Why do we need to copy Spark proto files ?  Can not add related spark-connect dependecnies ?

**Comment by pan3793** [comment]
> Why do we need to copy Spark proto files ? Can not add related spark-connect dependencies ?

@ulysses-you not like Hive, which provides a dedicated `hive-service-rpc` jar that only ships the generated vanilla thrift files, Spark ships the generated protobuf/gRPC files as part of `spark-connect-common_2.13` with Guava/Protobuf/gRPC class relocations.

For Kyuubi Commom and Server module, we just want a clean RPC layer, so copying proto files and regenerating is the cheapest way IMO.
